omc3/model/accelerators/lhc.py

```diff
@@ -486,12 +488,48 @@ def _get_madx_main_sequence_loading(self) -> str:
                 "has to be specified (--lhcmode option missing?)."
             )
 
-    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str]) -> str:
+    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str], update_dpp: bool = False) -> str:
         madx_script = self.get_base_madx_script()
-        for corr_file in corr_files:
+
+        # First set the dpp to the value in the accelerator model
+        madx_script += f"{ORBIT_DPP} = {self.dpp};\n"
+
+        for corr_file in corr_files:  # Load the corrections, can also update ORBIT_DPP
             madx_script += f"call, file = '{str(corr_file)}';\n"
-        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {self.dpp});\n"
+        
+        if update_dpp: # If we are doing orbit correction, we need to ensure that a correct and matching is done (in get_update_deltap_script)
```

Comment:
Probably the "and" here is a typo?

Reply:
I meant it in a way that the "correct" and the "match" needs to be done

Reply:
How about this [change](https://github.com/pylhc/omc3/pull/466/commits/63cd06cc546839fe0759e261faf9ba673ac938a6#diff-d1a4955c5b648d692ebecb63a82649e9614553aa5583a57647ee7995c81dc70fL500)

Reply:
Great, thanks